src/main/java/com/hubspot/jinjava/lib/tag/eager/EagerTagFactory.java

```diff
@@ -0,0 +1,37 @@
+package com.hubspot.jinjava.lib.tag.eager;
+
+import com.google.common.base.Throwables;
+import com.google.common.collect.ImmutableMap;
+import com.hubspot.jinjava.lib.tag.Tag;
+import java.util.Map;
+import java.util.Optional;
+
+public class EagerTagFactory {
+  public static final Map<Class<? extends Tag>, Class<? extends EagerTagDecorator<? extends Tag>>> EAGER_TAG_OVERRIDES = ImmutableMap
+    .<Class<? extends Tag>, Class<? extends EagerTagDecorator<?>>>builder()
+    .build();
+
+  @SuppressWarnings("unchecked")
+  public static <T extends Tag> Optional<EagerTagDecorator<T>> getEagerTagDecorator(
+    Class<T> clazz
+  ) {
```

Comment:
This will get called to register each tag in the `EagerExecutionMode.prepareContext()` (seen [here](https://github.com/HubSpot/jinjava/pull/543/files#diff-b251361da38e79a3eeee02262b427a58336d8666de114d39672f86b85e2fcae7R21))
@boulter This is the basic principle for how I am separating the eager execution code from the default execution code. The tags get wrapped with an `EagerTagDecorator` in this factory so that they can be registered to override the default tags